components/webext-storage/src/db.rs

```diff
@@ -105,6 +97,53 @@ impl DerefMut for StorageDb {
     }
 }
 
+// We almost exclusively use this SharedStorageDb
+pub struct SharedStorageDb {
+    db: Mutex<StorageDb>,
```

Comment:
Yep! `Store` always had some kind of external synchronization—that was `LazyStore` in Desktop, which kept it inside a `OnceCell<Mutex<Store>>`, or a `ConcurrentHandleMap` in the (hand-written 😩) FFI bindings—so it didn't need to be `Send + Sync`.

Pushing the synchronization down into this crate lets `Store` be `Send + Sync` for the transition to UniFFI later.

Reply:
Further, note that for uniffi'd components, the stores *already* are `Send + Sync` - eg, tabs didn't require any changes in this area because the fact that it's already exposed via uniffi meant the requirement already existed. To put it another way, if we were ever asked to expose webext-storage for use in Android via UniFFI, we'd be somewhat forced to make the store `Send + Sync` even without considering the bridged engine.